Kinvey.Core/Core/KinveyClientRequestInitializer.cs

```diff
@@ -105,10 +110,12 @@ public Credential KinveyCredential
 		/// </summary>
 		/// <param name="request">Request.</param>
 		/// <typeparam name="T">The response type of the request.</typeparam>
-        public void Initialize<T>(AbstractKinveyClientRequest<T> request)
+
+        public void Initialize<T>(AbstractKinveyClientRequest<T> request, string clientId = null)
```

Comment:
I think it depends. Right now, the only time we set the `AuthServiceID` is when we pass in a `clientId`, which is only in the `/oauth/token` case.  Since this initializer is used by all requests, do we want to have `AuthServiceID` be used for all basic authentication?

Reply:
Ok, fair enough. I thought each request got its own instance of `RequestInitializer`. Since that isn't the case, let's go with your implementation.